control/controlhttp/client.go

```diff
@@ -83,6 +89,159 @@ func (a *Dialer) httpsFallbackDelay() time.Duration {
 }
 
 func (a *Dialer) dial(ctx context.Context) (*controlbase.Conn, error) {
+	// If we don't have a dial plan, just fall back to dialing the single
+	// host we know about.
+	useDialPlan := envknob.BoolDefaultTrue("TS_USE_CONTROL_DIAL_PLAN")
```

Comment:
if you use an envknob like this then it won't be recorded earlier as in use and logged in the early logs which can help with debugging.

I might keep this here (for the DefaultTrue bit) but add a global like:

```go
var _ = envknob.RegisterBool("TS_USE_CONTROL_DIAL_PLAN") // to record at init time whether it's in use
```